arch-design/multicloud/BucketManagement.md

```diff
@@ -0,0 +1,369 @@
+
+Once the buckets are created, users should perform all the operations supported by soda-multicloud. For example: Uploading objects, setting lifecycle rules, migration etc.  
+
+
+![](resources/multi-cloud-fig1.png)
+
+
+![](resources/multi-cloud-fig2.png)
+
+
+
+#### API's(Create/Delete) to support bucket management:
+
+To support bucket managament, below is the list of API's with detailed description about parameters for various cloud vendors are below:
+
+
+#### AWS
+
+##### Rules for bucket naming:
+
+The following rules apply for naming S3 buckets:
+
+* Bucket names must be between 3 and 63 characters long.
+
+* Bucket names can consist only of lowercase letters, numbers, dots (.), and hyphens (-).
+
+* Bucket names must begin and end with a letter or number.
+
+* Bucket names must not be formatted as an IP address (for example, 192.168.5.4).
+
+* Bucket names can't begin with xn-- (for buckets created after February 2020).
+
+* Bucket names must be unique within a partition. A partition is a grouping of Regions. AWS currently has three partitions: aws (Standard Regions), aws-cn (China Regions), and aws-us-gov (AWS GovCloud [US] Regions).
+
+* Buckets used with Amazon S3 Transfer Acceleration can't have dots (.) in their names. For more information about transfer acceleration, see Amazon S3 Transfer Acceleration.
+
+
+##### URI Request Parameters
+
+The request uses the following URI parameters.
+
+`Bucket`: The name of the bucket to create.
+ Required: Yes
+
+`x-amz-acl`: The canned ACL to apply to the bucket.
+
+`Valid Values`: private | public-read | public-read-write | authenticated-read
+
+`x-amz-bucket-object-lock-enabled`: Specifies whether you want S3 Object Lock to be enabled for the new bucket.
+
+`x-amz-grant-full-control`: Allows grantee the read, write, read ACP, and write ACP permissions on the bucket.
+
+`x-amz-grant-read`: Allows grantee to list the objects in the bucket.
+
+`x-amz-grant-read-acp`: Allows grantee to read the bucket ACL.
+
+`x-amz-grant-write`: Allows grantee to create, overwrite, and delete any object in the bucket.
+
+`x-amz-grant-write-acp`: Allows grantee to write the ACL for the applicable bucket.
+
+##### API:
+***
+PUT / HTTP/1.1/BucketName
+***
+##### Request Body
+The request accepts the following data in XML format.
+
+`CreateBucketConfiguration`: Root level tag for the CreateBucketConfiguration parameters. Required: Yes
+
+`LocationConstraint`: Specifies the Region where the bucket will be created. If you don't specify a Region, the bucket is created in the US East (N. Virginia) Region (us-east-1).
+
+Type: String
+
+Valid Values: af-south-1 | ap-east-1 | ap-northeast-1 | ap-northeast-2 | ap-northeast-3 | ap-south-1 | ap-southeast-1 | ap-southeast-2 | ca-central-1 | cn-north-1 | cn-northwest-1 | EU | eu-central-1 | eu-north-1 | eu-south-1 | eu-west-1 | eu-west-2 | eu-west-3 | me-south-1 | sa-east-1 | us-east-2 | us-gov-east-1 | us-gov-west-1 | us-west-1 | us-west-2
```

Comment:
How are we getting these values? Are these values manually added as static or Generic across the multiple clouds? Do we include different cloud APIs to get this info Or just these static values?

Reply:
These are supported values for AWS. User has to provide the input. This list  is for reference and Example purpose only.